spotify_dl/spotify_dl.py

```diff
@@ -183,6 +192,7 @@ def spotify_dl():
             skip_non_music_sections=args.skip_non_music_sections,
             file_name_f=file_name_f,
             multi_core=args.multi_core,
+            proxy=args.proxy,
```

Comment:
the tests are failing because it is assumed that `-p` will always be set - if proxy is not set, then it will raise a KeyError

```
E               KeyError: 'proxy'
```

Reply:
I have no problem running without the 'p' option locally. I'm trying to figure out what's wrong but I can't find it. Maybe I should adapt the tests with the proxy option?

Reply:
The current error is because you try to fetch the key 'proxy' directly. This works when `-p` is set. However,  you can't expect everyone to be using the proxy, so it should work whether they use the proxy or not. The docs say:


https://github.com/yt-dlp/yt-dlp#network-options

> Pass in an empty string (--proxy "") for direct connection

So instead of doing 

```
 ydl_opts = {
                   "proxy": kwargs["proxy"],
```

You can do:

```
 ydl_opts = {
                   "proxy": kwargs.get('proxy')
```

The `.get()` function will return a None if the key doesn't exist and will prevent this error.

Reply:
Also, I realize I was pointing to the wrong line, so this should be applied below.

Reply:
Thank you for your explanation, I hadn't seen it that way. It should be correct from now on. 👍